### DISCUSSION TITLE
"head_errlog" features missing in master branch

### DISCUSSION THREAD

**Discussion by Loykos** [openzfs/zfs#14673] [Q&A] [answered]
Hi everyone,

I had my pool attached on my Mac OS X system (with 2.1.6 zfs version from this [repo](https://github.com/openzfsonosx)). It's ask me to upgrade the pool enabling "head_errlog" feature. So I did.. I have another Rhel9 home server with 2.1.9 version, so I didn't think it could be a problem.

When I try to import the pool from the rhel, I discovered that this feature it's not present in the 2.1.9 version, but it should be present only inside the master branch [(link)](https://openzfs.github.io/openzfs-docs/Basic%20Concepts/Feature%20Flags.html) (I don't know why this feature is inside the 2.1.6 osx version).

So I cloned the repo, checked out the master and built the package from source. Unfortunately if I execute the zpool executable inside the folder it still gives me that the "com.delphix:head_errlog" features is missing when I try to import the pool.

Is this feature inside the master or not? Which branch I should checkout?

`[loykos@localhost zfs]$ git branch`
`* master`

`[loykos@localhost zfs]$ ./zpool --version`
`zfs-2.1.99-1817_gb15ab50c4`
`zfs-kmod-2.1.9-1`

`[loykos@localhost zfs]$ sudo ./zpool import pool`
`This pool uses the following feature(s) not supported by this system:
	com.delphix:head_errlog (Support for per-dataset on-disk error logs.)
	org.openzfs:zilsaxattr (Support for xattr=sa extended attribute logging in ZIL.)
cannot import 'pool': unsupported version or feature`

Thanks!

**Comment by rincebrain** [comment, marked as answer]
You're running the 2.1.9 kernel modules, which don't have support for that. (cf. where it says `zfs-kmod-2.1.9-1` in the version output.)

**Reply by Loykos** [reply]
You're right. I solved installing latest 2.1.99 version kernel module. Thank you!
